src/lib/Form/Question/Question.tsx

```diff
@@ -17,153 +13,124 @@ import QuestionDocument from './Document/QuestionDocument';
+  const {lang = 'en', dateFormat = '', messages, readonly, triggerOnChange}: State = state
 
-    const props = {
+  const renderQuestion = (question: IQuestion) => {
+
+    const questionProps = {
       lang,
       question,
       messages,
       readonly,
       answer,
       isValid,
-      onChange: this.update,
+      onChange: update,
     };
     switch (question.question_type) {
       case QuestionType.TEXT:
-        return <QuestionText {...props}/>;
+        return <QuestionText {...questionProps}/>;
 
       case QuestionType.LONGTEXT:
-        return <QuestionLongText {...props}/>;
+        return <QuestionLongText {...questionProps}/>;
 
       case QuestionType.RADIO:
         if (question.possibilities.length < maxPossibilitiesBeforeAutocomplete)
-          return <QuestionRadio {...props}/>;
-        return <QuestionAutocomplete {...props}/>;
+          return <QuestionRadio {...questionProps}/>;
+        return <QuestionAutocomplete {...questionProps}/>;
       //
       case QuestionType.SELECT:
         if (question.possibilities.length < maxPossibilitiesBeforeAutocomplete)
-          return <QuestionSelect {...props}/>;
-        return <QuestionAutocomplete {...props}/>;
+          return <QuestionSelect {...questionProps}/>;
+        return <QuestionAutocomplete {...questionProps}/>;
 
       case QuestionType.CHECKBOX:
         if (question.possibilities.length < maxPossibilitiesBeforeAutocomplete)
-          return <QuestionCheckbox {...props}/>;
-        return <QuestionAutocomplete multiSelect {...props}/>;
+          return <QuestionCheckbox {...questionProps}/>;
+        return <QuestionAutocomplete multiSelect {...questionProps}/>;
 
       case QuestionType.DATE:
-        if (this.props.dateFormat)
-          return <QuestionDate {...props} dateFormat={this.props.dateFormat}/>;
-        return <QuestionText {...props}/>;
+        if (dateFormat)
+          return <QuestionDate {...questionProps} dateFormat={dateFormat}/>;
+        return <QuestionText {...questionProps}/>;
 
       case QuestionType.DOCUMENT:
-        return <QuestionDocument {...props}/>;
+        return <QuestionDocument {...questionProps}/>;
 
       case QuestionType.LABEL:
         return '';
 
       default:
-        return <QuestionText {...props}/>;
+        return <QuestionText {...questionProps}/>;
     }
   }
 
-  componentWillUnmount() {
-    const {question, updateSectionValidity, removeAnswer, removeCheckedPossibility} = this.props;
-    updateSectionValidity(question.section_id, question.id, true);
-    removeAnswer(question.id);
-    if (isDependable(question)) {
-       removeCheckedPossibility(question.id);
+  useEffect(() => {
+    return () => {
+      dispatch(updateSectionValidity(question.section_id, question.id, true));
+      dispatch(removeAnswer(question.id));
+      if (isDependable(question)) {
+        dispatch(removeCheckedPossibility(question.id));
+      }
     }
-  }
+  }, [])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has missing dependencies: 'dispatch', 'question', 'removeAnswer', 'removeCheckedPossibility', and 'updateSectionValidity'. Either include them or remove the dependency array.](https://app.codacy.com/gh/Particeep/form-answer/pullRequest?prid=9527160)
```suggestion
  }, [dispatch, question, removeAnswer, removeCheckedPossibility, updateSectionValidity])
```